Makefile

```
@@ -222,3 +235,26 @@ swagger-docs: ## preview the API documentation
 		-e 'REDOC_OPTIONS=hide-hostname="true" lazy-rendering' \
 		-p $(SWAGGER_DOCS_PORT):80 \
 		bfirsh/redoc:1.6.2
+
+.PHONY: buildx
+ifeq ($(BUILDX), bundles/buildx)
+buildx: bundles/buildx # build buildx cli tool
+else
+buildx:
+endif
+
+bundles/buildx: GOOS ?= $(shell go env GOHOSTOS || uname | awk '{print tolower($$0)}' || echo windows)
+bundles/buildx: GOARCH ?= $(shell go env GOHOSTARCH || echo amd64)
+bundles/buildx: DOCKER_BUILDX_VERSION ?= v0.3.0
+bundles/buildx: DOCKER_BUILDX_REPO ?= https://github.com/docker/buildx.git
+bundles/buildx: BUILDX_DOCKERFILE ?= Dockerfile.buildx
+bundles/buildx: bundles ## build buildx CLI tool
+	# This intetionally is not using the `--output` flag from the docker CLI which is a buildkit option
+	# The idea here being that if buildx is being used, it's because buildkit is not supported natively
+	docker build -f $(BUILDX_DOCKERFILE) -t "moby-buildx:$(DOCKER_BUILDX_VERSION)" \
+		--build-arg BUILDX_COMMIT="$(DOCKER_BUILDX_VERSION)" \
```

instead of passing the makefile variable here, could we change these to;

```suggestion
		--build-arg BUILDX_COMMIT \
```

That way;

- if no`BUILDX_COMMIT` environment variable is set, the default from the Dockerfile is used
- if `BUILDX_COMMIT` is set as environment variable, it overrides the default in the Dockerfile
- we can simplify the Makefile and remove the steps above that set the defaults